src/WebJobs.Script.WebHost/Security/FileSystemSecretsRepository.cs

```diff
@@ -135,6 +135,16 @@ private void OnChanged(object sender, FileSystemEventArgs e)
             }
         }
 
+        public async Task<string[]> GetSecretFilesByPrefix(string prefix)
```

Comment:
If the purge logic is not updated, all your non-decryptable files will end up being deleted.